### DISCUSSION TITLE
How to set the params of DynUnet?

### DISCUSSION THREAD

**Discussion by markvvw** [Project-MONAI/MONAI#5033] [Q&A]
I know the function that calculates params, but it gives us a wrong result. So I come here for a correct way to calculate kernels and strides.

in_channel=1
out_channel=1
ima_size=128*128
spacing=1*1

thanks

**Comment by Nic-Ma** [comment]
Hi @yiheng-wang-nv ,

Could you please help share some best practices for the DynUNet?

Thanks in advance.

**Reply by markvvw** [reply]
> Hi @yiheng-wang-nv ,
> 
> Could you please help share some best practices for the DynUNet?
> 
> Thanks in advance.

我在做图像分割，但如果使用transformer相关的网络，一是占显存，二是收敛慢，所以选择CNN; 最近才发现dynunet, 可以加残差和深监督，想试试，但这个kernels strides计算，真的很头疼，输出结果多一个维度，头疼

**Reply by markvvw** [reply]
> Hi @yiheng-wang-nv ,
> 
> Could you please help share some best practices for the DynUNet?
> 
> Thanks in advance.

Thanks in advance.

**Reply by markvvw** [reply]
> Hi @yiheng-wang-nv ,
> 
> Could you please help share some best practices for the DynUNet?
> 
> Thanks in advance.

其实我主要想问的是 DynUnet的深监督怎么用？已经看了官网说明，需要torch.unbind，但没弄明白怎么使用，能给我提供一个关于这里的代码展示么？谢谢

**Reply by yiheng-wang-nv** [reply]
Hi @markvvw , an example of how to use dynunet for decathlon dataset is prepared, please refer to: https://github.com/Project-MONAI/MONAI/blob/b1aa40dc70ab6e2f75ddf3327da97102bdd7546a/monai/networks/nets/dynunet.py#L89

As for the kernel size and strides part, you can refer to this [function](https://github.com/Project-MONAI/tutorials/blob/main/modules/dynunet_pipeline/create_network.py#L8)

**Comment by neuronflow** [comment]
I struggle a bit to wrap my head around this...

How to best set the parameters of DynUnet to approximate (especially `kernel_size` and `strides`):

```
model = BasicUNet(
    dimensions=3,
    in_channels=1,
    out_channels=1,
    features=(32, 32, 64, 128, 256, 32),
    )
```

Thanks for your time!

**Reply by yiheng-wang-nv** [reply]
Hi @neuronflow , [this is one method](https://github.com/Project-MONAI/tutorials/blob/main/modules/dynunet_pipeline/create_network.py#L8) to determine the kernel size and strides, you can start from it to build your baseline.

**Reply by neuronflow** [reply]
thanks, I saw that. In our case, we want to walk the other route. We already have a trained model with BasicUNet, now we want to see how DynUnet compares to it. Therefore, we want to use a model with similar parameters.
